Energy_Preserving.jl

```diff
@@ -0,0 +1,500 @@
+    tree = rootedtree(array)
+    l = length(array)
+    superarray = get_permutations(array)
+    lperm = length(superarray)
+    for i in reverse(1:lperm) 
+        if superarray[i][1] != 1
+            splice!(superarray, i)
+        end
+    end
+    lperm = length(superarray)
+    flag = false
+    for i in reverse(1:lperm)
+        for j in 1:(l-1)
+            if (superarray[i][j+1] - superarray[i][j]) > 1
+                flag = true
+            end
+        end
+        if flag == true
+            splice!(superarray, i)
+        end
+        flag = false
+    end
+    lperm = length(superarray)
+    for i in reverse(1:lperm)
+        arbol = rootedtree(superarray[i])
+        bandera = arbol == tree
+        if bandera == false
+            splice!(superarray, i)
+        end
+    end
+    bandera = false
+    lperm = length(superarray)
+    for i in reverse(1:lperm)
+        for j in 1:(i-1)
+            if superarray[i] == superarray[j]
+                bandera = true
+            end
+        end 
+        if bandera == true
+            splice!(superarray, i)
+        end
+        bandera = false
+    end
+
+    return superarray
+end
+
+function get_permutations(arr)
+    perms = collect(permutations(arr))
+    return perms
+end
+
+function num_leafs(a)
+    k = a[end]
+    return k - 1
+end
+
+function add_one_to_left(arrays)
+    for i in 1:length(arrays)
+        arrays[i] = vcat([1], arrays[i])
+    end
+    return arrays
+end
+
+#this function multplies the coefficient for its symmetry factor
+function symfact_normalization(coef,thetrees)
+    l = length(coef)
+    for i in 1:l
+        factor = 0
+        factor = RootedTrees.symmetry(RootedTree(thetrees[i]))
+        coef[i] = coef[i]*(1//factor)
+    end
+    return coef
+end
+
+#this function tells up to what order a method is Energy Preserving
+function OrderMethod(A,b)
+    s = 1
+    flag = false
+    rka = RungeKuttaMethod(A, b)
+    while flag == false 
+#generate bseries 
+        series_a = modified_equation(bseries(rka, s))
+        coefficients = collect(values(series_a))
+        atrees = collect(keys(series_a))
+# Create an empty vector to store the converted trees
+        trees = Vector{Vector{Int}}(undef, length(series_a))
+# Convert the trees and store them in the trees vector
+        for i in 1:length(series_a)
+            array_from_RTree = root_converter(atrees[i])
+            if isempty(array_from_RTree)
+                trees[i] = Int[]
+            else
+                trees[i] = array_from_RTree
+            end
+        end
+        coefficients = symfact_normalization(coefficients,trees)  
+        #println(trees)
+        #println(coefficients)
+        if IsEnergyPreserving(trees,coefficients) == true
+            flag = true
+        end
+        s = s + 1
+    end
+    println("Energy Preserving for order < ", s-1)
+end
+
+#this function checks if an AVF method is energy Preserving for a given order s
+function EnergyPreservingAVF(s) 
```

Comment:
It would be good to have a general interface of such a function that accepts a B-series. This implementation looks quite similar to `EnergyPreserving(A,b,s)`. If you have a single implementation accepting a B-series, you can re-use it for RK methods and the AVF method (after generating their corresponding B-series).